pulp_2to3_migration/app/plugin/content.py

```diff
@@ -407,61 +408,62 @@ def get_remote_by_importer_id(importer_id):
                         downloaded=pulp2content.downloaded
                     )
                     if artifact is None:
```

Comment:
i am pretty sure i have screwed up the indentation  since at least the counter sduring the migration got off, but the gist of this change is that we cannot skip when artifact is None due to the content that has futures, those futures need to be resolved otherwise we would get into the problem where in one batch there would be manifests+tags( in the ContentSaverStage), and since tags rely on manifests being already saved at the tags _pre_save_hook() tags won't have needed information at the save() time.

Reply:
we also need to check with @quba42  on how this change will impact debian pipeline.
I have checked rpm one - it does have futures -Rpms, that are getting resolved and then in the interrelate stage relation betwen modules and rpms are created. For the case when a modular rpm would miss, there would be created one relation less. tldr - rpm plugin is not impacted.

Reply:
I am not 100% sure.
Here is what I do rely on: Later types in the order given by the ordered dict `DebMigrator.content_models`, may query for the objects created by earlier types in the order from their `create_pulp3_content()` method. e.g.:
```
release = pulp3_models.Release.objects.filter(<some_uniquely_identifying_filter>).first()
```
Currently, if that query returns `None`, the pipeline *will* fail.

I understand I need to add graceful handling for the case that this query will return `None` because the target content was never created because it's artifact was corrupt, and it was therefore skipped.

I am not sure if there can also be a case were the query returns none because the target content has not yet passed sufficiently far through the pipeline, which would be a bigger problem for the Debian pipeline...

Reply:
it should not be the case because debian package is a future type content and will be resolved/saved/skipped if corrupted/ by the time PackageReleaseComponent content will be saved.